### PR TITLE
Fix for Issue #542

### PR DESCRIPTION
**Pull request by Avd6977**
Skip downloading the file if it is downloaded server side.

### CODE REVIEW

**Comment by Avd6977** [comment]
@gabrielliwerant Any chance this PR could make the version bump with the custom filters?

**Comment by gabrielliwerant** [comment]
@Avd6977 I don't think so, but I can do a smaller release after this next one for some quick fixes.

**Comment by gabrielliwerant** [comment]
Does this PR actually answer the #542 issue? Looks like that user was referring to an issue where only the current page is downloaded with serverside, not extra downloads.

**Comment by Avd6977** [comment]
That'd be awesome, thank you!

It provides one possible answer to it, which is to just download their own file. The other answer is to get all of the data serverside using "onDownload", which is existing functionality.

**Comment by Avd6977** [comment]
@gabrielliwerant Just curious when you think that smaller release may be.

**Comment by gabrielliwerant** [comment]
This feature probably won't make next release, in case you were wondering. I need some time to think about it, and how it may interact with an earlier PR that is also vying to change download options.  There are also some more critical things that should take priority.

**Comment by Avd6977** [comment]
Ok, would you prefer it be in a different manner? Our use case is that the entire CSV is downloaded server-side but we'd like to still use the download button.

**Comment by gabrielliwerant** [comment]
I'm reluctant to extend the API for specialized use cases unless there's a strong case for it and it can't be done in another manner. What if instead the `onDownload` callback were enabled to skip downloading if it returned `false`? This way, we don't need a new option, and it can be utilized in a wider variety of contexts.

**Comment by Avd6977** [comment]
I'm good with that. Committed the code change. Let me know if that looks like what you were talking about.

File: test/MUIDataTableToolbar.test.js [hunk 1]
```diff
@@ -241,4 +240,30 @@ describe('<TableToolbar />', function() {
 
     assert.strictEqual(onDownload.callCount, 1);
   });
+
+  it('should trigger onDownload prop callback when calling method handleCSVDownload and skip download if return from onDownload is false', () => {
```
Comment (gabrielliwerant) [inline test/MUIDataTableToolbar.test.js]: This test doesn't prove that the download is skipped, so I'm not sure it's doing anything for us that the other test for `onDownload` isn't already doing. I think that the way the CSV creation works at the moment, it's difficult to determine in a test whether or not the download was successful (just whether or not the callback fired). To do so, we'd probably need to expose another callback that fires after successful download and returns the data, but this is outside the scope of this PR, and I'm not sure we need it at this point.

Long story short, I appreciate the attempt to add to our coverage, but I think it would better to remove this test for now.

Instead, if you want, we could add to the `csv-export` example a conditional that would trigger this option and prevent download, so at least we'd have a built in way to verify in the codebase. If you don't want to do that, that's fine, I can add it after this gets merged in, before the next release.

Comment (Avd6977) [reply test/MUIDataTableToolbar.test.js]: Removed the test and added `onDownload: () => false` to the example!

File: examples/csv-export/index.js [hunk 2]
```diff
@@ -62,6 +62,7 @@ class Example extends React.Component {
           filename: 'excel-format.csv',
           separator: ';',
       },
+      onDownload: () => false,
```
Comment (gabrielliwerant) [inline examples/csv-export/index.js]: So, the only issue with this is that it defaults to off for people testing csv export functionality in the example. I was thinking something like this might be more instructive:

```
state = { isOnDownload: true };

...

const options = { onDownload: () => this.state.isOnDownload };

...

<button onClick={ () => this.setState({ isOnDownload: false }) }>Shut off download functionality</button>
```

Does that make sense?

Comment (Avd6977) [reply examples/csv-export/index.js]: Yep, makes sense. Updated and pushed!

Comment (gabrielliwerant) [reply examples/csv-export/index.js]: Thank you!
